### PR TITLE
api: fix revisions resolving

### PR DESCRIPTION
**Pull request by pazembrz**
*Fixes revisions resolving when there are gaps in revisions numbers

*INSPIR-3486

### CODE REVIEW

**Comment by pazembrz** [comment]
It's quite important patch for Inspire. Without it revisions are broken for us. @inveniosoftware/architects

**Comment by michamos** [comment]
For context, the issue is that `record.commit()` increments the `version_id` (and hence the `revision_id`), but that version of the record only gets persisted when calling `db.session.commit()`. So if you call `record.commit()` twice, there will be gaps in the `version_id`s in the version table, and hence gaps in `revision_id`.

So the two following properties can't hold at once: 
1. `list(record.revisions)[n] == record.revisions[n]` for all `n` with `0 <= n < len(list(record.revisions))`
2. `record.revisions[n].revision_id == n` for all  `n` for which a `revision_id` exists.

This commit chooses to guarantee property 2. which is more useful (you can get a revision easily knowing given its `revision_id`). For negative `n`, in order to preserve backwards compatibility it behaves as property 1. because property 2. doesn't make sense for negative `n`, `revision_id` being always positive.

**Comment by lnielsen** [comment]
Good catch! We're reviewing right now and will fast-track a release including backports.

File: tests/test_api.py [hunk 1]
```diff
@@ -182,6 +182,46 @@ def test_revisions(app, db):
...
+def test_retrieve_proper_revision(app, db):
+    """Check accessing revision with gaps
+
+    Test checks if it's possible to access proper revision
+    when revision numbers have 'gaps'
+    """
+    record = Record.create({'title': 'test 1'})
+    db.session.commit()
+    record['title'] = "test 2"
+    record.commit()
```
Comment (slint) [inline tests/test_api.py:194]: I think another interesting side-effect to add as a test case is calling `record.commit()` multiple times, without a `db.session.commit()` in between. In this case, I believe the revision number gets bumped but no extra rows end up in the DB (which I guess is also probably the reasoning behind this PR).

**Comment by pazembrz** [comment]
I've added another commit which adds `__reversed__` method for `reversed` operator to work after `__getitem__` was changed

**Comment by lnielsen** [comment]
Note, a v1.3.2 release is coming as soon as #222 is green

**Pull request merged**